examples/llava/clip.h

```diff
@@ -26,6 +27,23 @@ extern "C" {
 
 struct clip_ctx;
 
+// RGB uint8 image
+struct clip_image_u8 {
+    int nx;
+    int ny;
+
+    std::vector<uint8_t> buf;
+};
+
+// RGB float32 image (NHWC)
+// Memory layout: RGBRGBRGB...
+struct clip_image_f32 {
+    int nx;
+    int ny;
+
+    std::vector<float> buf;
+};
+
```

Comment:
The CLIP api should be C-style - it cannot use C++ containers

Reply:
resolved, wasn't sure the convention, I haven't worked in a lot of mixed C/C++ codebases

is there some better way to declare the structs are they are now shared across 3 files? feels messy

Reply:
The idea is to keep the image structs oblique (i.e. the user (in this case `server`) should not be aware what is the actual data representation inside them) and update the C-style API with whatever functionality is needed to access those.

As a basic example, instead of writing:

```c++
img_sl.img_data->buf = image_buffer;
```

We should have something like:

```c++
clip_image_u8_set(img_sl.img_data, image_buffer.data());
```

Reply:
Got it this makes sense. 

It seems it's possible to avoid `img_sl.img_data->buf = image_buffer;` entirely through exposing `llava_image_embed_make_with_clip_img` in `llava.h`. This I believe is the cleanest set of changes. Though maybe I am missing something in exposing this function.

However if this implementation doesn't satisfy I have two others in mind.
1. From the previous code add something like `clip_image_u8_set` to avoid struct direct access. This definitely can be done, but felt a bit hacky as we are actually using the same buf twice for different purposes. First to store the raw image data, and then the clip image data. 
2. Add `std::vector<uint8_t> buf;` to `slot_image` in server/utils.hpp. Decode the base64 image directly into it, and then call `llava_image_embed_make_with_bytes` on that buf. This would be instead of reusing the same buf for different purposes which was done in the last set of changes

Reply:
This should be fine for now. The CLIP/LLaVA code can be improved in many ways, but likely a complete rewrite would be best

Could you fix the Zig build before merging